web/docs/auth/overview.md

```diff
@@ -382,45 +397,52 @@ export const updatePassword = async (args, context) => {
 <TabItem value="ts" label="TypeScript">
 
 ```ts title="src/server/actions.ts"
+import {
+    createProviderId,
+    findAuthIdentity,
+    updateAuthIdentityProviderData,
+    deserializeAndSanitizeProviderData,
+} from '@wasp/auth/utils.js';
 import type { UpdatePassword } from '@wasp/actions/types'
-import type { User } from '@wasp/entities'
-
-type UpdatePasswordPayload = {
-  userId: User['id']
-}
 
 export const updatePassword: UpdatePassword<
-  UpdatePasswordPayload,
-  User
+  { email: string; password: string },
+  void,
 > = async (args, context) => {
-  return context.entities.User.update({
-    where: { id: args.userId },
-    data: {
-      password: 'New pwd which will be hashed automatically!',
-    },
+  const providerId = createProviderId('email', args.email)
+  const authIdentity = await findAuthIdentity(providerId)
+  if (!authIdentity) {
+      throw new HttpError(400, "Unknown user")
+  }
+  
+  const providerData = deserializeAndSanitizeProviderData<'email'>(authIdentity.providerData)
+
+  // Updates the password and hashes it automatically.
+  await updateAuthIdentityProviderData(providerId, providerData, {
+      hashedPassword: args.password,
   })
 }
 ```
 
 </TabItem>
 </Tabs>
 
-### Default validations
+### Default Validations
 
 When you are using the default authentication flow, Wasp validates the fields with some default validations. These validations run if you use Wasp's built-in [Auth UI](/docs/auth/ui) or if you use the provided auth actions.
 
 If you decide to create your [custom auth actions](/docs/auth/username-and-pass#2-creating-your-custom-sign-up-action), you'll need to run the validations yourself.
 
 Default validations depend on the auth method you use.
 
-#### Username & password
+#### Username & Password
 
 If you use [Username & password](/docs/auth/username-and-pass) authentication, the default validations are:
 
 - The `username` must not be empty
 - The `password` must not be empty, have at least 8 characters, and contain a number
 
-Note that `username`s are stored in a **case-sensitive** manner.
+Note that `username`s are stored in a **case-insensitive** manner.
```

Comment:
Since we are normalising the provider user ID equally, the behaviour of the user case sensitivity changed (for the better iMHO)

Reply:
Yes, this is better!